src/svc_ioq.c

```diff
@@ -257,6 +257,7 @@ svc_ioq_flushv(SVCXPRT *xprt, struct xdr_ioq *xioq)
 			xioq->frag_hdr_bytes_sent += result;
 			iov[0].iov_base += result;
 			iov[0].iov_len -= result;
+			frag_hdr_size -= result;
```

Comment:
Is this going to cause a problem down below when frag_hdr_size is subtracted from result?

Reply:
> Is this going to cause a problem down below when frag_hdr_size is subtracted from result?

sorry, I checked the code, and did not find this will cause a problem. actually, the variable related to frag_hdr_size is remaining, fbytes and xioq->write_start, all need frag_hdr_size is subtracted from result.

Reply:
> Good point. Maybe, instead, use `if (result < (frag_hdr_size - xioq->frag_hdr_bytes_sent))`

if we only use `if (result < (frag_hdr_size - xioq->frag_hdr_bytes_sent))` to replace `if (result < frag_hdr_size)`.  but without the change of this pull request, the problem still exist.

Reply:
But if the whole frag header isn't sent, we goto again, to try and send the rest of it.

Below, when frag_hdr_size is used, it's after all the attempts to send, so we would only subtract the reduced frag_hdr_size from remaining and fbytes, and add to write_start. But since we've sent the whole frag header, we should change all of those by the original frag_hdr_size

Reply:
@ffilz thanks the reply,  remaining and fbytes always record the left body length, it should not subtract the frag header. in the second time we send 5bytes(result=5) including the left 2bytes frag header and 3bytes body, then remaining should subtract 3 bytes.

Reply:
Okay, I agree.

Reply:
OK, I see it now. Can we have some comments talking about what's going on here because it is really confusing as evidenced by my confusion throughout this discussion. Maybe an example showing the two packets, one with part of the fragment header, and the other with the rest of the fragment header and some portion of the fragment.

Other than that, I approve of this change now that I understand it.

Reply:
sure, already add some comments.